tracker/tracker/src/main/app/session.ts

```diff
@@ -77,7 +77,7 @@ export default class Session {
 
   setMetadata(key: string, value: string) {
     this.metadata[key] = value
-    this.handleUpdate({ metadata: { [key]: value } })
+    this.handleUpdate({ metadata: { ...this.metadata } })
```

Comment:
`{ ...this.metadata }` => `metadata: this.metadata` pretty much the same with less operations

but read parent comment anyways

Reply:
I deliberatly created a copy of the object to avoid issues if this.metadata was modified before handleUpdate is using it. The cost is minimal anyway